### ISSUE TITLE
EntryPointNotFoundException for CurrentlyRunningVisualStudioDiffTool raised on OSX

### ISSUE THREAD

**Issue by mihaipopescu** [shouldly/shouldly#466]
Any test that uses a Shoudly function raises an exception if the project targets .Net Framework (4.7) and uses mono.

No exception is thrown for .Net Core 2.0.

```
System.TypeInitializationException : The type initializer for 'Shouldly.ShouldlyConfiguration' threw an exception.
  ----> System.TypeInitializationException : The type initializer for 'Shouldly.Configuration.KnownDiffTools' threw an exception.
  ----> System.EntryPointNotFoundException : CreateToolhelp32Snapshot
  at Shouldly.ShouldBeTestExtensions.ShouldBe[T] (T actual, T expected, System.Func`1[TResult] customMessage) [0x0000e] in <f125d08698344cd4a060acb31b3fea32>:0 
  at Shouldly.ShouldBeTestExtensions.ShouldBe[T] (T actual, T expected) [0x00001] in <f125d08698344cd4a060acb31b3fea32>:0 
  at Unity.Foo.Tests.Bar.Parse_ParameterValue () [0x00013] in /Users/mihai/Development/unity/Unity.Foo/Unity.Foo.Tests/Bar.cs:52 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00032] in /private/tmp/source-mono-2017-04/bockbuild-2017-04/profiles/mono-mac-xamarin/build-root/mono-x64/mcs/class/corlib/System.Reflection/MonoMethod.cs:305 
--TypeInitializationException
  at Shouldly.Configuration.DiffToolConfiguration..ctor () [0x0000b] in <f125d08698344cd4a060acb31b3fea32>:0 
  at Shouldly.ShouldlyConfiguration..cctor () [0x00000] in <f125d08698344cd4a060acb31b3fea32>:0 
--EntryPointNotFoundException
  at (wrapper managed-to-native) Shouldly.Configuration.CurrentlyRunningVisualStudioDiffTool+ParentProcessUtils:CreateToolhelp32Snapshot (uint,uint)
  at Shouldly.Configuration.CurrentlyRunningVisualStudioDiffTool+ParentProcessUtils.ParentProcess (System.Diagnostics.Process process) [0x0000a] in <f125d08698344cd4a060acb31b3fea32>:0 
  at Shouldly.Configuration.CurrentlyRunningVisualStudioDiffTool+ParentProcessUtils+<GetSelfAndAncestors>d__2.MoveNext () [0x0008a] in <f125d08698344cd4a060acb31b3fea32>:0 
  at System.Collections.Generic.LargeArrayBuilder`1[T].AddRange (System.Collections.Generic.IEnumerable`1[T] items) [0x0005e] in /private/tmp/source-mono-2017-04/bockbuild-2017-04/profiles/mono-mac-xamarin/build-root/mono-x64/external/corefx/src/Common/src/System/Collections/Generic/LargeArrayBuilder.cs:142 
  at System.Collections.Generic.EnumerableHelpers.ToArray[T] (System.Collections.Generic.IEnumerable`1[T] source) [0x00033] in /private/tmp/source-mono-2017-04/bockbuild-2017-04/profiles/mono-mac-xamarin/build-root/mono-x64/external/corefx/src/Common/src/System/Collections/Generic/EnumerableHelpers.cs:86 
  at System.Linq.Enumerable.ToArray[TSource] (System.Collections.Generic.IEnumerable`1[T] source) [0x00015] in /private/tmp/source-mono-2017-04/bockbuild-2017-04/profiles/mono-mac-xamarin/build-root/mono-x64/external/corefx/src/System.Linq/src/System/Linq/ToCollection.cs:19 
  at Shouldly.Configuration.CurrentlyRunningVisualStudioDiffTool.GetPath () [0x00006] in <f125d08698344cd4a060acb31b3fea32>:0 
  at Shouldly.Configuration.CurrentlyRunningVisualStudioDiffTool..ctor () [0x00000] in <f125d08698344cd4a060acb31b3fea32>:0 
  at Shouldly.Configuration.KnownDiffTools..ctor () [0x000c6] in <f125d08698344cd4a060acb31b3fea32>:0 
  at Shouldly.Configuration.KnownDiffTools..cctor () [0x00000] in <f125d08698344cd4a060acb31b3fea32>:0
```

Could probably be related with #461 but in this case we're reusing the same dll and not attempt to rebuild a new one.

**Comment by JosephWoodward** [comment]
Thanks for reporting this. What version of Shouldly are you using out of interest?

**Comment by mihaipopescu** [comment]
Thanks for the quick reply. We're using version 2.8.3.

**Comment by JosephWoodward** [comment]
This should be fixed in the latest release ([3.0.0-beta0003](https://www.nuget.org/packages/Shouldly/3.0.0-beta0003))

**Comment by mihaipopescu** [comment]
Thanks Joseph. And just as a note, this was not happening in 2.8.2.
